extension/src/Explorer/testExplorer.ts

```diff
@@ -26,20 +27,47 @@ export class TestExplorer implements TreeDataProvider<TestTreeNode> {
 
     public getTreeItem(element: TestTreeNode): TreeItem {
         return {
-            label: this.getFriendlyElementName(element),
-            collapsibleState: element.isFolder ? TreeItemCollapsibleState.Collapsed : void 0,
+            label: element.name,
+            collapsibleState: element.isMethod ? TreeItemCollapsibleState.None : TreeItemCollapsibleState.Collapsed,
             command: this.getCommand(element),
             iconPath: this.getIconPath(element),
-            contextValue: element.level.toString(),
+            contextValue: element.type.toString(),
         };
     }
 
-    public getChildren(element?: TestTreeNode): TestTreeNode[] | Thenable<TestTreeNode[]> {
-        if (element) {
+    public getChildren(element?: TestTreeNode): TestTreeNode[] | Promise<TestTreeNode[]> {
+        if (!element) {
+            const folders = workspace.workspaceFolders;
+            return folders.map((folder) => new TestTreeNode(folder.name, folder.name, TestTreeNodeType.Folder, folder.uri.toString()));
+        } else if (!element.children) {
+            return new Promise(async (resolve: (res: TestTreeNode[]) => void): Promise<void> => {
+                const results: SearchResults[] = await FetchTestsUtility.searchTestEntries({
+                    nodeType: element.type,
+                    uri: element.uri,
+                    fullName: element.fullName,
+                });
+                const parentSuite: TestSuite = this.toTestSuite(element);
+                if (parentSuite) {
+                    const childSuites: TestSuite[] = results.map((result) => result.suite);
+                    for (const childSuite of childSuites) {
+                        childSuite.parent = parentSuite;
+                    }
+                    parentSuite.children = childSuites;
+                    this.updateTestStorage(childSuites);
+                }
+                element.children = results.map((result) => new TestTreeNode(
+                    result.displayName,
+                    result.suite.test,
+                    result.nodeType,
+                    result.suite.uri,
+                    result.suite.range,
+                    element,
+                ));
+                resolve(element.children);
+            });
+        } else {

```

Comment:
Wrap less code in control blocks. There are 3 branches and the third one can be at the second place. So the current second one can be out of the control block. Code is generally easier to read when there are less control blocks.
```ts
if (!element) {
  ...
} else if (element.children) {
  ...
}

return new Promise(async...);
```